scripts/staging/kubectl-oci.sh

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+# Manage kubectl plugin binaries as OCI artifacts.
+# Supports both pushing and pulling operations.
+# Usage:
+#   Push: ./kubectl-oci.sh push --tag <tag> --namespace <namespace> --username <user> --password <token>
+#   Pull: ./kubectl-oci.sh pull --tag <tag> --namespace <namespace> --username <user> --password <token>
+
+set -euo pipefail
+
+SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]:-"$0"}")")"
+ROOT_DIR="$SCRIPT_DIR/../"
+
+source "$ROOT_DIR/utils/log.sh"
+
+ACTION=""
+TAG=""
+NAMESPACE=""
+USERNAME=""
+PASSWORD=""
+REGISTRY="ghcr.io"
+PLUGIN="${PLUGIN:-"mayastor"}"
+
+usage() {
+  cat << EOF
+Usage: $0 <action> [options]
+
+Actions:
+  push    Push kubectl binaries to $REGISTRY as OCI artifacts
+  pull    Pull kubectl binaries from $REGISTRY OCI artifacts
+
+Options:
+  --tag <tag>             Release tag (required)
+  --registry <registry>   The registry to push/pull from [default=$REGISTRY]
+  --namespace <namespace> Namespace path (required)
+  --username <username>   Registry username (required)
+  --password <password>   Registry token/password (required)
+
+Examples:
+  $0 push --tag v1.0.0 --namespace $PLUGIN/dev --username user --password token
+  $0 pull --tag v1.0.0 --namespace $PLUGIN/dev --username user --password token
+EOF
+}
+
+parse_args() {
+  if [[ $# -lt 1 ]]; then
+    usage
+    log_fatal "Error: Action required (push or pull)"
+  fi
+
+  ACTION="$1"
+  shift
+
+  case "$ACTION" in
+    push|pull)
+      ;;
+    *)
+      usage
+      log_fatal "Error: Invalid action '$ACTION'. Must be 'push' or 'pull'"
+      ;;
+  esac
+
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+      --tag)
+        TAG="$2"
+        shift 2
+        ;;
+      --namespace)
+        NAMESPACE="$2"
+        shift 2
+        ;;
+      --registry)
+        REGISTRY="$2"
+        shift 2
+        ;;
+      --username)
+        USERNAME="$2"
+        shift 2
+        ;;
+      --password)
+        PASSWORD="$2"
+        shift 2
+        ;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      *)
+        usage
+        log_fatal "Unknown option: $1"
+        ;;
+    esac
+  done
+
+  if [[ -z "$TAG" ]] || [[ -z "$NAMESPACE" ]] || [[ -z "$USERNAME" ]] || [[ -z "$PASSWORD" ]]; then
+    usage
+    log_fatal "Error: All options (--tag, --namespace, --username, --password) are required"
```

Comment:
can we make user and pass optional in case we are already logged in? Also avoids typing a password in the command line